rustls/src/lib.rs

```diff
@@ -5,6 +5,16 @@
 //! <!-- (as tracked in: https://github.com/brody4hire/portable-rustls/issues/31) -->
 //! __IMPORTANT NOTICE:__ regardless of upstream __`rustls`__ project this fork is __NOT CERTIFIED__ and __NOT PEER-REVIEWED__ - USE AT YOUR OWN RISK
 //!
+//! ## REQUIREMENTS
+//!
+//! <!-- TODO(portable-rustls) CLEANUP & IMPROVE DOC FOR THIS FORK -->
+//! Requirements for building with this fork with default build configuration, which uses `Arc` from `portable-atomic-util`:
+//!
+//! - Use Rust nightly toolchain
+//! - in case of no-std: add `portable-atomic` with `critical-section` or `unsafe-assume-single-core` feature enabled - see the following for more info & requirements: <https://docs.rs/portable-atomic/latest/portable_atomic/#optional-features>
+//! - possibly more requirements in case of `portable-atomic` with `critical-section` for no-std: <https://docs.rs/critical-section/latest/critical_section/#usage-in-no-std-binaries>
+//! - need to build with some kind of crypto provider for this TLS functionality to be useful in general
```

Comment:
- looks like this has some old info with requirement to add `portable-atomic`, this is no longer needed with the new `critical-section` & `unsafe-assume-single-core` features
- may be best to remove this, restore section for targets wtih no atomic ptr below, and update the info in that whole section according to these updates